src/CMakeLists.txt

```diff
@@ -171,6 +171,9 @@ endif(CCTAG_NO_COUT)
 if(CCTAG_VISUAL_DEBUG)
   target_compile_definitions(CCTag PRIVATE "-DCCTAG_VISUAL_DEBUG")
 endif(CCTAG_VISUAL_DEBUG)
+if(HAVE_SHFL_DOWN_SYNC)
+  target_compile_definitions(CCTag PRIVATE "-DCCTAG_HAVE_SHFL_DOWN_SYNC")
```

Comment:
this should go only in the case if `CCTAG_WITH_CUDA` is ON, as it can be only active on cuda code, am I right?
If so, it should be move up at line 116
```cmake
  target_compile_definitions(CCTag
                         PUBLIC -DCCTAG_WITH_CUDA -DCUB_CDP
                         PRIVATE ${TBB_DEFINITIONS})

  if(CCTAG_HAVE_SHFL_DOWN_SYNC)
    target_compile_definitions(CCTag PRIVATE "-DCCTAG_HAVE_SHFL_DOWN_SYNC")
  endif()
```

Reply:
Hmm. The generated cctag_config.hpp has so far been ignored in the CCTag code. I understand that it's relevant for distributing CCTag but isn't it reasonable to use the same file inside CCTag as well as soon as we start using it?

Assuming that the answer is yes, I've added an include_directories() so that it is actually visible during compilation. But of course, when we start using cctag_config, then it's redundant to have target_compile_definitions as well.

Reply:
yes the `cctag_config.hpp` is ignored in the code because everything is managed by cmake (it propagates the definitions even when the library is used as third pary like in AliceVision).
But nevertheless, it is better to have for people that may not use cmake as building system.
At least they can include that before any other cctag header.
So I agree that it is not much useful in our configuration but it's better to have in the installed version.
For this reason please remove the  `include_directories()` because we don't need that to be visible. FYI, the correct way to include that would be using 
```cmake
target_include_directories(<target> PUBLIC             
            $<BUILD_INTERFACE:${generated_dir}>  #this will include the generated dir *only* for in-tree building
            $<INSTALL_INTERFACE:include/> #this will include the the path where config.hpp is installed for using as third party
```

Reply:
BTW, the same holds for the version.hpp file, it is installed but never used, at least for the moment. In case we will need it in the code in the future then, again, we have to add that `target_include_directories` piece of code.

Reply:
I think we should start using our own generated config files at some point. The CXXFLAGS can in projects using CCTags can be manipulated as shown here: [cmake-remove-a-compile-flag-for-a-single-translation-unit](https://stackoverflow.com/questions/28344564/cmake-remove-a-compile-flag-for-a-single-translation-unit).
I'll try to remember that no-so-simple target_include_dir statement from above.